pkg/pulp_client/artifacts.go

```diff
@@ -24,3 +26,39 @@ func (r *pulpDaoImpl) LookupArtifact(ctx context.Context, sha256sum string) (*st
 		return readResp.Results[0].PulpHref, fmt.Errorf("fetched artifact with sha256sum %v, expected at most 1 result, but got %v", sha256sum, len(readResp.Results))
 	}
 }
+
+func (r *pulpDaoImpl) ListArtifacts(ctx context.Context, offset, limit int32) (artifacts []zest.ArtifactResponse, total int, err error) {
+	ctx, client := getZestClient(ctx)
+	resp, httpResp, err := client.ArtifactsAPI.ArtifactsList(ctx, r.domainName).Fields([]string{"file", "sha256"}).Limit(limit).Offset(offset).Execute()
+	if httpResp != nil {
+		defer httpResp.Body.Close()
+	}
+	if err != nil {
+		return artifacts, 0, errorWithResponseBody("error listing artifacts", httpResp, err)
+	}
+	return resp.Results, int(resp.Count), err
+}
+
+func (r *pulpDaoImpl) ListAllArtifactSHA256s(ctx context.Context) ([]string, error) {
+	initial := int32(0)
+	limit := int32(300)
+
+	artifacts, total, err := r.ListArtifacts(ctx, initial, limit)
+	if err != nil {
+		return nil, err
+	}
+	for len(artifacts) < total {
+		initial += limit
+		artifactList, _, err := r.ListArtifacts(ctx, initial, limit)
+		if err != nil {
+			return nil, err
+		}
+		artifacts = append(artifacts, artifactList...)
+	}
```

Comment:
the artifact list api has a sha256 filter:  https://pulpproject.org/pulpcore/restapi/#tag/Artifacts/operation/artifacts_list which should make this a lot more efficient

Reply:
oh, it sadly doesn't support a list of sha256 hashes, but you're right it probably should be more efficient then this, will change 👍🏼